float32[6] x
float32[6] y
float32[6] th
